File: 백준/Silver/3758. KCPC/KCPC.java
import java.io.*;
import java.util.*;
class Main{
    public static void main(String[] args)throws Exception{
        BufferedReader I=new BufferedReader(new InputStreamReader(System.in));
    
        StringBuilder as=new StringBuilder();
        int ti=Integer.parseInt(I.readLine());
        while(ti-->0){
            StringTokenizer st=new StringTokenizer(I.readLine());
            int n=Integer.parseInt(st.nextToken()),
            k=Integer.parseInt(st.nextToken()),
            t=Integer.parseInt(st.nextToken())-1,
            m=Integer.parseInt(st.nextToken());
            int[][] tsc=new int[n][k],team=new int[n][4];
            for(int i=0;i<n;i++){
                team[i][3]=i;
            }
            for(int mi=0;mi<m;mi++){
                st=new StringTokenizer(I.readLine());
                int i=Integer.parseInt(st.nextToken())-1,
                j=Integer.parseInt(st.nextToken())-1,
                s=Integer.parseInt(st.nextToken());
                if(s>tsc[i][j]) tsc[i][j]=s;
                team[i][1]++;
                team[i][2]=mi;
            }
            for(int i=0;i<n;i++){
                for(int j:tsc[i]){
                    team[i][0]+=j;
                }
            }
            Arrays.sort(team,new Comparator<int[]>() {
                public int compare(int[] a,int[] b){
                    if(a[0]!=b[0]) return b[0]-a[0];
                    if(a[1]!=b[1]) return a[1]-b[1];
                    return a[2]-b[2];
                }
            });
            for(int i=0;i<n;i++){
                if(t==team[i][3]) as.append(i+1).append('\n');
            }
        }
        System.out.print(as);
    }
}

File: 백준/Silver/3758. KCPC/KCPC.py
import sys
def input(): return sys.stdin.readline().rstrip()
op=[]

for ti in range(int(input())):
    n,k,t,m=map(int,input().split())
    t-=1
    team=[[0]*4 for _ in range(n)]
    tsc=[[0]*k for _ in range(n)]
    for i in range(n):
        team[i][3]=i
    for mi in range(m):
        i,j,s=map(int,input().split())
        i-=1
        j-=1
        if s>tsc[i][j]:
            tsc[i][j]=s
            
        team[i][1]+=1
        team[i][2]=mi
        
    for i in range(n):
        for j in tsc[i]:
            team[i][0]+=j
    team.sort(key=lambda x:(-x[0],x[1],x[2]))
    for i in range(n):
        if t==team[i][3]:
            op.append(str(i+1))
    
print('\n'.join(op))